spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -67,6 +67,24 @@ public void setOutputChannelName(String outputChannelName) {
 		this.outputChannelName = outputChannelName;
 	}
 
+	/**
+	 * Return the output channel.
+	 * @return the channel.
+	 * @since 4.3
+	 */
+	@Override
+	public MessageChannel getOutputChannel() {
+		if (this.outputChannelName != null) {
+			synchronized (this) {
+				if (this.outputChannelName != null) {
+					this.outputChannel = getChannelResolver().resolveDestination(this.outputChannelName);
+					this.outputChannelName = null;
```

Comment:
:smile: I don't know English, so what to say about French.
But I see why we haven't fixed that before. We resolved `outputChannelName` in the `start()`, not `afterPropertiesSet()`.
Although this your change is more robust for the `runtime model`: we might not start the component before the the `GraphServer` logic.
So, LGTM!